Regression/Checksum/backend/amrex_backend.py

```diff
@@ -0,0 +1,80 @@
+"""
+Copyright 2021
+
+This file is part of WarpX
+
+License: BSD-3-Clause-LBNL
+"""
+
+import re
+import numpy as np
+import yt
+from yt.frontends.boxlib.data_structures import AMReXDataset
+
+yt.funcs.mylog.setLevel(50)
+
+class Backend:
+    ''' Use AMReXDataset as the backend reader to read AMReX plotfiles
+    '''
+
+    def __init__(self, filename):
+        ''' Constructor: store the dataset object
+        '''
+
+        self.dataset = AMReXDataset(filename)
+
+    def fields_list(self):
+        ''' Return the list of fields defined on the grid
+        '''
+
+        return [item for item in self.dataset.field_list if item[0] == 'boxlib']
+
+    def species_list(self):
+        ''' Return the list of species in the dataset
+        '''
+
+        return set([item[0] for item in self.dataset.field_list if
+                            item[1][:9] == 'particle_' and item[0] != 'all'])
+
+    def n_levels(self):
+        ''' Return the number of MR levels in the dataset
+        '''
+
+        return self.dataset.max_level+1;
+
+    def get_field_checksum(self, lev, field, test_name):
+        ''' Calculate the checksum for a given field at a given level in the dataset
+        '''
+
+        # Boolean variable parsed from test name needed for workaround below
+        galilean = True if re.search('galilean', test_name) else False
```

Comment:
For debugging, could you print `test_name` here, as well as the value of `Galilean`? Maybe, for some reason, the `test_name` was changed when we refactored this checksum library.

Reply:
OK that didn't help, `galilean` has the right value.